src/fsharp/vs/Reactor.fs

```diff
@@ -83,9 +82,14 @@ type Reactor() =
                         Trace.TraceInformation("Reactor: --> wait for background (debug only), remaining {0}, mem {1}, gc2 {2}", inbox.CurrentQueueLength, GC.GetTotalMemory(false)/1000000L, GC.CollectionCount(2))
                         match bgOpOpt with 
                         | None -> ()
-                        | Some bgOp -> 
-                            while bgOp ctok do 
-                                ()
+                        | Some bgOp ->
+                            let rec loop () =
+                                async {
+                                    let! r = bgOp ctok
```

Comment:
An observation: the ``Step`` background action (which is what ``bgOp`` is) is never actually cancelled in the code - it's always only called as part of the Reactor agent, and the core async loop of the MailboxProcessor is never cancelled (effectively runs with CancellationToken.None).

This just makes me wonder whether either
1. it would be useful to make thee individual steps cancellable, or
2. if the ``bgOp`` should be non-Async to record the invariant that it is never cancelled.  

I'm not sure, need to think